### DISCUSSION TITLE
Error: Multiple Partitions in a table (without having multiple partitions)

### DISCUSSION THREAD

**Discussion by yeykawb** [TabularEditor/TabularEditor3#336] [🙋♀️TE3 Q&A]
Hey,

I'm wondering what the cause is for TE3 returning this error when trying to update the model schema. 
I can update each table individually without issues and as you can see, the analyzer returns only 1 partition for each table in the model.

/  Jakob

![image](https://user-images.githubusercontent.com/97282358/148518766-b149897a-929a-47af-a636-9a6c8c21cf22.png)

**Comment by otykier** [comment]
The [Use Analysis Services for change detection](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services) feature creates a temporary table on the server, when updating the table schema. This table is created with 1 partition for each table that you want to update the schema for. This allows Tabular Editor to collect schema information from multiple tables in one go. Now unfortunately, we did not consider that some editions of Analysis Services do not support tables with more than one partition, so we will have to add some kind of fallback mechanism that automatically updates them one-by-one in cases where multiple partitions are not supported.

**Reply by yeykawb** [reply]
I don't really follow on the above why this maneuver doesn't work (as it has worked like a charm since I started using TE3). Also, no table has more than 1 partition, at least from what I can see. The only change has been me adding some manual M-code tables (still 1 partition).

/ Jakob

**Reply by otykier** [reply]
If you were able to perform a schema update on multiple tables previously, then most likely the M code you added contains some logic that Tabular Editor doesn't "understand", which causes it to fall back to using Analysis Services for change detection. You can test this by multi-selecting a number of tables excluding the one where you edited the M code recently (hold down CTLR to select tables one-by-one). In that case, it would be helpful to know what the M code you added/edited was, as we would like to provide support for more M functions than what we have today (Tabular Editor only natively supports a small subset of M functions).

But in any case, we have a bug in that Tabular Editor should not generate a temporary schema detection table with more than 1 partition, when falling back to Analysis Services change detection while connected to an edition of Analysis Services that does not support tables with multiple partitions.

**Reply by yeykawb** [reply]
Ok, thanks. In that case it has got to do with the M-code. 

When the following code is used:

![image](https://user-images.githubusercontent.com/97282358/148534987-66ccc81a-6335-43aa-90ea-8b1a028409d7.png)

TE allows me to set the data type with M-code. However, when multiselecting tables and updating schema, TE returns that I have multiple partitions. When single-selecting this table with the above M-code, it returns no issues with the schema detection.

I've used a simpler version of this M-code earlier, without declaring data types as described in this [blog-post](https://blog.crossjoin.co.uk/2016/06/03/creating-tables-in-power-bipower-query-m-code-using-table/).

![image](https://user-images.githubusercontent.com/97282358/148535530-55daf5d8-6e9e-4567-90eb-16ec34fd96ad.png)

But I wanted to declare one column as a number. When trying to update the above by using this code, and updating the schema:
![image](https://user-images.githubusercontent.com/97282358/148535448-68cabaa1-40db-4691-95b4-7ca76ca69572.png)

TE says that the column is deleted (this code however works in Visual Studio).

Therefore, in an attempt to get data type declared I resorted to the first code I posted:
![image](https://user-images.githubusercontent.com/97282358/148534987-66ccc81a-6335-43aa-90ea-8b1a028409d7.png)

And it worked - until I tried to update the schema in bulk. Does this make sense to you?

**Reply by otykier** [reply]
Yes - thanks a lot! We'll look into it.
